packages/bitcore-lib/lib/script/interpreter.js

```diff
@@ -307,33 +419,37 @@ Interpreter.prototype.initialize = function(obj) {
   this.altstack = [];
   this.pc = 0;
   this.satoshis = 0;
-  this.sigversion = Interpreter.SIGVERSION_BASE;
+  this.sigversion = Signature.Version.BASE;
   this.pbegincodehash = 0;
   this.nOpCount = 0;
   this.vfExec = [];
   this.errstr = '';
   this.flags = 0;
+  this.execdata = {};
 };
 
 Interpreter.prototype.set = function(obj) {
   this.script = obj.script || this.script;
   this.tx = obj.tx || this.tx;
-  this.nin = typeof obj.nin !== 'undefined' ? obj.nin : this.nin;
+  this.nin = typeof obj.nin === 'undefined' ? this.nin : parseInt(obj.nin);
   this.stack = obj.stack || this.stack;
   this.altstack = obj.altack || this.altstack;
```

Comment:
Is this a typo? 'obj.altack' - instead of 'obj.altstack'

Reply:
Yes